### PR TITLE
Adding default PGD commit_scopes for BigAnimal 

### PR DESCRIPTION
**Pull request by jpe442**

### CODE REVIEW

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+deepToC: true
+---
+
+## Commit scope
+
+[Commit scopes](../../../pgd/latest/durability/commit-scopes/) in PGD is a set of rules that describes the behavior of the system as transactions are committed. Because they define how transactions are replicated across a distributed database, they impact consistency and reliability.
+
+The actual behavior depends on which kind of commit scope a commit scope's rule uses: [Group Commit](/../../../pgd/latest/durability/group-commit/), [Commit At Most Once](/../../../pgd/latest/durability/camo/), [Lag Control](/../../../pgd/latest/durability/lag-control/), [Synchronous Commit](/../../../pgd/latest/durability/synchronous_commit/), or a combination of these.
+
+Customizing these settings can be complex, presenting a challenge for users.
```
Comment (djw-m) [inline product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:13]: Not keen on this language. Suggests to the reader they need to be up to the task.

"This flexibility means that selecting a balanced combination of rules can take time. To speed up deployment, BigAnimal's PGD has a preset selection of commit_scopes for typical user requirements."

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx [hunk 2]
```diff
@@ -0,0 +1,113 @@
+---
+title: Postgres Distributed (PGD) presets on BigAnimal
+navTitle: PGD presets on BigAnimal
+deepToC: true
+---
+
+## Commit scope
+
+[Commit scopes](../../../pgd/latest/durability/commit-scopes/) in PGD is a set of rules that describes the behavior of the system as transactions are committed. Because they define how transactions are replicated across a distributed database, they impact consistency and reliability.
```
Comment (djw-m) [inline product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:9]:
```suggestion
[Commit scopes](../../../pgd/latest/durability/commit-scopes/) in PGD are a set of rules that describe the behavior of the system as transactions are committed. Because they define how transactions are replicated across a distributed database, they have an effect on consistency, durability and performance.
```

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+The actual behavior depends on which kind of commit scope a commit scope's rule uses: [Group Commit](/../../../pgd/latest/durability/group-commit/), [Commit At Most Once](/../../../pgd/latest/durability/camo/), [Lag Control](/../../../pgd/latest/durability/lag-control/), [Synchronous Commit](/../../../pgd/latest/durability/synchronous_commit/), or a combination of these.
+
+Customizing these settings can be complex, presenting a challenge for users. 
+
+### BigAnimal's commit_scope preset options
+
+BigAnimal's commit scope presets simplify managing commit scopes in PGD. These presets balance customization with ease of use, allowing users to enjoy PGD's benefits without the complexity of manual configurations:
```
Comment (djw-m) [inline product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:17]: This para reads more like marketing :) Better to present in brief the four options

BigAnimal's commit scope presets balance customization with ease of use and reduce the options down to four presets.

The presets include a rule for high consistency within a subgroup and a separate rule for lag control. There are then combinations of these rules for high consistency within a subgroup and lag control for other subgroups, and high coniositency within the subgroup and with one or more nodes of an other subgroups.

Something along those lines.

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    rule := 'MAJORITY (location-a) SYNCHRONOUS_COMMIT',
+    wait_for_ready := true
+);
+```
+
+The rule says that for transactions originating from a node in `location-a`, the majority of the nodes in `location-a` must acknowledge the transaction before it is committed.
+
+ba001 uses `MAJORITY` instead of `ALL` so that in the case of 3 data nodes, the third can be updated asynchronously. This allows for a node failure without interrupting a single region service. 
+
+This is the baseline BigAnimal provided commit_scope and is the commit_scope for all new sub-groups.
```
Comment (djw-m) [inline product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:47]: Surely default commit scope.

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+To make a commit_scope the `default_commit_scope` for a node group, use the following command, `bdr.alter_node_group_option`:
+
+```sql
+SELECT bdr.alter_node_group_option(
+  node_group_name := '<node_group_name>',
+  config_key := 'default_commit_scope',
+  config_value := '<commit_scope_name>'
+);
+```
+
```
Comment (djw-m) [inline product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:112]: Might be an idea to mention that commit scopes can be applied per transaction and that the BA presets do not stop users creating their own commit scopes which can be applies as needed.

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+| Preset ID | Local group         | External groups     | Use Case                             |
+|-----------|---------------------|--------------------|--------------------------------------|
+| ba001     | MAJORITY, SYNCHRONOUS_COMMIT  | N/A                | High consistency for critical data   |
+| ba002     | ALL, LAG_CONTROL         | ALL, LAG_CONTROL        | Manages replication lag locally and externally |
+| ba003     | MAJORITY, SYNCHRONOUS_COMMIT  | ANY 1, LAG_CONTROL        | Combines high consistency locally with lag control for external|
+| ba004     | MAJORITY, SYNCHRONOUS_COMMIT  | ANY 1, SYNCHRONOUS COMMIT | Higher consistency via external validation |
+
+!!! Note
+ba001 is also the only BigAnimal preset that works with single-grouped PGD clusters. All of them, including ba001, work with multi-grouped clusters.
```
Comment (djw-m) [inline product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:27]:
```suggestion
ba001 is also the only BigAnimal preset that works with PGD clusters that have only one data group. All of them, including ba001, work with PGD clusters that have wo or more data groups.
```

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+!!!
+
+### ba001
+
+ba001 offers a default setup that optimizes data consistency and reliability across distributed environments. Here is an example definition for a group of nodes, `location-a`:
```
Comment (djw-m) [inline product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:32]: Execpt none of these commit scopes appear like this.

e.g.

```
bdrdb=> select * from bdr.commit_scopes ;
 commit_scope_id |  commit_scope_name   | commit_scope_origin_node_group |                                                                                commit_scope_rule
-----------------+----------------------+--------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
      2651729099 | ba001_p-ghhbcdvrnm-a |                     3588189531 | MAJORITY (p-ghhbcdvrnm-a) SYNCHRONOUS_COMMIT
      3685052456 | ba002_p-ghhbcdvrnm-a |                     3588189531 | ALL (p-ghhbcdvrnm-a) LAG CONTROL (max_commit_delay=1s, max_lag_size=100MB) AND ALL NOT (p-ghhbcdvrnm-a) LAG CONTROL (max_lag_size=1000MB, max_commit_delay=4s, max_lag_time=30s)
       552598214 | ba003_p-ghhbcdvrnm-a |                     3588189531 | MAJORITY (p-ghhbcdvrnm-a) SYNCHRONOUS_COMMIT AND ANY 1 NOT (p-ghhbcdvrnm-a) LAG CONTROL (max_lag_size=1000MB, max_commit_delay=4s, max_lag_time=30s)
      4133704097 | ba004_p-ghhbcdvrnm-a |                     3588189531 | MAJORITY (p-ghhbcdvrnm-a) SYNCHRONOUS_COMMIT AND ANY 1 NOT (p-ghhbcdvrnm-a) SYNCHRONOUS_COMMIT
(4 rows)
```

You may want to add notes about how commit scopes also include the name of the subgroup they are referring to (as this would be the origin node group).

**Review by djw-m — COMMENTED** [review]
Comments left.

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+BigAnimal's commit scope presets simplify managing commit scopes in PGD. These presets balance customization with ease of use, allowing users to enjoy PGD's benefits without the complexity of manual configurations:
+
+| Preset ID | Description                         | Use Case                             |
+|-----------|-------------------------------------|--------------------------------------|
+| ba001     | SYNCHRONOUS_COMMIT                  | High consistency for critical data   |
+| ba002     | LAG_CONTROL + NOT                   | Manages replication lag with external validation|
+| ba003     | SYNCHRONOUS_COMMIT + LAG_CONTROL + NOT   | Combines consistency with lag control and external validation|
+| ba004     | SYNCHRONOUS_COMMIT + NOT            | Combines consistency with external validation |
+
+### BigAnimal's commit_scope catalog
```
[the remark this answers is not in the document]
Comment (jpe442) [reply product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx]: done

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx (hunk 1 above)
Comment (jpe442) [reply product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:13]: subbed this in 👍

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx (hunk 3 above)
Comment (jpe442) [reply product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:17]: Cool, got it.

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx (hunk 4 above)
Comment (jpe442) [reply product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:47]: 👍

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx (hunk 5 above)
Comment (jpe442) [reply product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:112]: cool. Added the first suggestion toward the end and the second suggestion toward the beginning.

File: product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx (hunk 7 above)
Comment (jpe442) [reply product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx:32]: I am not following this 100%. Can you expand?

 I took a stab anyway if you want to check it out in the newest push.

Comment (djw-m) [reply product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx]: ba0001 will only be a prefix to a commit scope
The full name of the actual commit scope will be ba001_<subgroupname>

Someone could come away with the expectation of seeing a commit scope called ba001 when there is none.

Comment (jpe442) [reply product_docs/docs/biganimal/release/using_cluster/pgd_defaults_ba.mdx]: got it.

**Review by djw-m — APPROVED** [review]
LGTM (have we had a BA tech review?)

**Comment by jpe442** [comment]
> LGTM (have we had a BA tech review?)

Yes Aaron Sonntag checked it out and put in his suggestions. I also checked in #eng-pgd regarding the wording describing the rules.
